src/Skia/Perspex.Skia/FormattedTextImpl.cs

```diff
@@ -75,36 +108,206 @@ public Size Measure()
+                            }
+                        }
+                    }
+
+
+                }
+
+                PerspexFormattedTextLine line = new PerspexFormattedTextLine();
+                line.Start = curOff;
+                line.Length = measured;
+                line.Width = lineWidth;
+                line.Height = lineHeight;
+                line.Top = curY;
+
+                if (line.Width < 0)
+                    line.Width = _skiaRects[line.Start + line.Length - 1].Right;
+
+                //Build character rects
+                for (int i = line.Start; i < line.Start + line.Length; i++)
+                {
+                    float prevRight = 0;
+                    if (i != line.Start)
+                        prevRight = _skiaRects[i - 1].Right;
+
+                    subString = _text.Substring(line.Start, i - line.Start + 1);
+                    float w = Paint.MeasureText(subString);
+
+                    SKRect rc;
+                    rc.Left = prevRight;
+                    rc.Right = w;
+                    rc.Top = line.Top;
+                    rc.Bottom = line.Top + line.Height;
+                    _skiaRects[i] = rc;
+                }
+
+                subString = _text.Substring(line.Start, line.Length);
+                line.Width = Paint.MeasureText(subString);
+
+                _skiaLines.Add(line);
+
+                curY += lineHeight; // + mLeading;
+                curOff += measured + extraSkip;
+            }
+
+            // Now convert to Perspex data formats
             _lines.Clear();
             _rects.Clear();
             float maxX = 0;
-            for (var c = 0; c < _shared->LineCount; c++)
+
+            for (var c = 0; c < _skiaLines.Count; c++)
             {
-                var w = _shared->Lines[c].Width;
+                var w = _skiaLines[c].Width;
                 if (maxX < w)
                     maxX = w;
-                _lines.Add(new FormattedTextLine(_shared->Lines[c].Length, _shared->Lines[c].Height));
+
+                _lines.Add(new FormattedTextLine(_skiaLines[c].Length, _skiaLines[c].Height));
             }
+
             for (var c = 0; c < _text.Length; c++)
-                _rects.Add(_shared->Bounds[c].ToRect());
-            if (_shared->LineCount == 0)
+                _rects.Add(_skiaRects[c].ToPerspexRect());
+
+            if (_skiaLines.Count == 0)
                 _size = new Size();
             else
             {
-                var lastLine = _shared->Lines[_shared->LineCount - 1];
+                var lastLine = _skiaLines[_skiaLines.Count - 1];
                 _size = new Size(maxX, lastLine.Top + lastLine.Height);
             }
         }
 
-        void Rebuild()
+        internal void Draw(SKCanvas canvas, SKPoint origin)
         {
-            MethodTable.Instance.RebuildFormattedText(Handle);
-            Reload();
+            SKPaint paint = Paint;
+
+            /* This originated from Native code, it might be useful
```

Comment:
If this commented-out code is to be kept, can we at least get a better explanation of why it might be useful?